src/access/for.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { UrlString, WebId } from "../interfaces";
+import { internal_defaultFetchOptions } from "../resource/resource";
+import {
+  Access,
+  getAgentAccess,
+  getGroupAccess,
+  getPublicAccess,
+} from "./universal";
+
+// Note: The module's name is "for", because it exports "*AccessFor" methods, and
+// it is imported as "access/for".
+
+export type Actor = "agent" | "group" | "public";
+
+/**
+ * Get an overview of what access is defined for a given actor (Agent, Group or everyone).
+ *
+ * This function works with Solid Pods that implement either the Web Access
+ * Control spec or the Access Control Policies proposal, with some caveats:
+ *
+ * - If access to the given Resource has been set using anything other than the
+ *   functions in this module, it is possible that it has been set in a way that
+ *   prevents this function from reliably reading access, in which case it will
+ *   resolve to `null`.
+ * - It will only return access specified explicitly for the given Agent. If
+ *   additional restrictions are set up to apply to the given Agent in a
+ *   particular situation, those will not be reflected in the return value of
+ *   this function.
```

Comment:
I _think_ that every overload can take its own docblock. So maybe what we should do here is repeat this doc block for the first two overloads (so that the `@param`s are accurate), and then make sure that this refers to either "Agent/Group" or "everyone", as appropriate.

Reply:
(For extra bonus points we could even split up the `agent` and `group` overloads so that one takes a `WebId` and the other a `UrlString`, but since that's a lot of copy-pasting and both resolve to a string anyway, I'd be totally happy not to do that.)